File: app/src/main/java/com/channa/mobiledatausageapp/view/BaseActivity.java
package com.channa.mobiledatausageapp.view;

import android.view.View;

import com.channa.mobiledatausageapp.R;
import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;
import butterknife.BindView;
import butterknife.ButterKnife;

public abstract class BaseActivity extends AppCompatActivity {

    @BindView(R.id.parentLayout)
    View parentLayout;

    protected void initViews() {
        ButterKnife.bind(this);
    }

    protected void showErrorActionSnackBar(String message) {
        final Snackbar snackbar = Snackbar.make(parentLayout, message, Snackbar.LENGTH_INDEFINITE);
        snackbar.setAction("Dismiss", new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                snackbar.dismiss();
            }
        }).setActionTextColor(getResources().getColor(android.R.color.holo_red_light)).show();
    }
}

File: app/src/main/java/com/zak/datauses/views/BaseActivity.kt
package com.zak.datauses.views

import android.view.View
import androidx.appcompat.app.AppCompatActivity
import butterknife.BindView
import butterknife.ButterKnife
import com.zak.datauses.R
import com.google.android.material.snackbar.Snackbar

abstract class BaseActivity : AppCompatActivity() {
    @BindView(R.id.parentLayout)
    var parentLayout: View? = null
    protected open fun initViews() {
        ButterKnife.bind(this)
    }

    protected fun showErrorActionSnackBar(message: String?) {
        val snackbar = Snackbar.make(parentLayout!!, message!!, Snackbar.LENGTH_INDEFINITE)
        snackbar.setAction("Dismiss") { snackbar.dismiss() }.setActionTextColor(resources.getColor(android.R.color.holo_red_light)).show()
    }
}